elixir/apps/domain/lib/domain/directories.ex

```diff
@@ -0,0 +1,23 @@
+defmodule Domain.Directories do
+  alias Domain.Repo
+  alias Domain.Directories.Provider
+  alias Domain.Accounts
+  alias Domain.Auth
+
+  def fetch_provider_by_id(id) do
+    Provider.Query.all()
```

Comment:
Are we planning on completely removing an auth provider record from the DB when they are deleted?

If not, is there ever a reason we'd need to reference a soft deleted auth provider?